src/PrestaShopBundle/Controller/Admin/Improve/International/CurrencyController.php

```diff
@@ -500,6 +565,14 @@ private function getErrorMessages(Exception $e)
                     'This currency already exists.',
                     'Admin.International.Notification'
                 ),
+                CurrencyConstraintException::EMPTY_BULK_TOGGLE => $this->trans(
+                    'Currencies must be provided in order to toggle their status',
+                    'Admin.International.Notification'
+                ),
+                CurrencyConstraintException::EMPTY_BULK_DELETE => $this->trans(
+                    'At least one currency must be provided for deleting',
```

Comment:
Can you please explain in which context do those exceptions appear? Is it when a user is performing an action but without having selected a currency?

Reply:
> Is it when a user is performing an action but without having selected a currency?

Exactly !

Reply:
Noted, thanks! Symfony pages don't allow bulk actions (delete or enable/disable) when no item is selected, see screenshot below.

<img width="326" alt="Capture d’écran 2020-05-26 à 18 22 37" src="https://user-images.githubusercontent.com/32565890/82925433-fea25000-9f7d-11ea-8128-22608bf0154e.png">

On the Legacy pages, the error notification is _You must select at least one element to delete._. So I suggest keeping the Symfony behavior. :-)

Reply:
Okay ! It's good for me then ... Thanks @LouiseBonnard

Reply:
@LouiseBonnard it is indeed blocked on the front side, so this error shouldn't happen But it's better to manage it in case the template is overridden or modified by a module It's better to have a safeguard on the server side as well
So as you said the appropriate wording would be the same as in legacy `You must select at least one element to delete.` in `Admin.Notifications.Error` domain?
What about the the toggle bulk action, I couldn't find a wording for this one, and since we don't know if the action was enable or disable maybe we could have a generic wording like `You need to select at least one element to use the bulk action.` (in `Admin.Notifications.Error` as well)

Reply:
Well, it's a bit inconsistent to talk about 'bulk action' and make it works for a single item... but I've checked and it happens for all grids in the software so let's keep on being consistent in our inconsistency, aha! So okay for:

1. _You must select at least one element to delete._ in _Admin.Notifications.Error_
1. _You must select at least one element to perform a bulk action._ in _Admin.Notifications.Error_

Many thanks, @jolelievre & @PululuK. :-)

Reply:
Thanks @LouiseBonnard, But ... we need two messages, one when no item is selected during mass deletion and the other when changing status. Otherwise, a single common message for the two cases.

Reply:
_You must select at least one element to perform a bulk action._ works for both cases: bulk delete action and bulk disable/enable action. What do you think?

Reply:
LGTM  thanks @LouiseBonnard !  @jolelievre WDYF ?